### PR TITLE
Add table azure_key_vault_deleted_vault. Closes #262

### LINKED ISSUE #262: Add table `azure_key_vault_deleted_vault`
**Is your feature request related to a problem? Please describe.**
https://docs.microsoft.com/en-us/rest/api/keyvault/vaults/list-deleted#deletedvaultlistresult

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/azure_key_vault_deleted_vault []

PRETEST: tests/azure_key_vault_deleted_vault

TEST: tests/azure_key_vault_deleted_vault
Running terraform
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 5s [id=/subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/resourceGroups/turbottest559]
azurerm_key_vault.named_test_resource: Creating...
azurerm_key_vault.named_test_resource: Still creating... [10s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [20s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [30s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [40s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [50s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [1m0s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [1m10s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [1m20s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [1m30s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [1m40s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [1m50s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [2m0s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [2m10s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [2m20s elapsed]
azurerm_key_vault.named_test_resource: Still creating... [2m30s elapsed]
azurerm_key_vault.named_test_resource: Creation complete after 2m36s [id=/subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/resourceGroups/turbottest559/providers/Microsoft.KeyVault/vaults/turbottest559]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "az keyvault delete --name turbottest559 --resource-group turbottest559"]
null_resource.named_test_resource: Still creating... [10s elapsed]
null_resource.named_test_resource (local-exec): WARNING: Warning! If you have soft-delete protection enabled on this key vault, you will not be able to reuse this key vault name until the key vault has been purged from the soft deleted state. Please see the following documentation for additional guidance.
null_resource.named_test_resource (local-exec): https://docs.microsoft.com/en-us/azure/key-vault/general/soft-delete-overview
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "az keyvault list-deleted > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/azure_deleted_key_vault/terraform/test/output.json"]
null_resource.named_test_resource: Still creating... [20s elapsed]
null_resource.named_test_resource: Creation complete after 21s [id=7932574734936461083]
data.local_file.input: Reading...
data.local_file.input: Read complete after 0s [id=9da9adfbc9e91e2a0ee2104bcf32fd29b121b8fa]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

region = "eastus"
resource_aka = "azure:///subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983"
resource_aka_lower = "azure:///subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/microsoft.keyvault/locations/eastus/deletedvaults/turbottest79983"
resource_id = "/subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983"
resource_name = "turbottest79983"
subscription_id = "d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983",
    "name": "turbottest79983",
    "region": "eastus",
    "resource_group": "turbottest79983",
    "subscription_id": "d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8",
    "type": "Microsoft.KeyVault/deletedVaults"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983",
    "name": "turbottest79983",
    "type": "Microsoft.KeyVault/deletedVaults"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983",
      "azure:///subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/microsoft.keyvault/locations/eastus/deletedvaults/turbottest79983"
    ],
    "name": "turbottest79983",
    "title": "turbottest79983"
  }
]
✔ PASSED

POSTTEST: tests/azure_key_vault_deleted_vault

TEARDOWN: tests/azure_key_vault_deleted_vault

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  id,
  type,
  deletion_date,
  scheduled_purge_date
from
  azure_key_vault_deleted_vault;
+-----------------+---------------------------------------------------------------------------------------------------------------------------------+----------------------------------+--------------------
| name            | id                                                                                                                              | type                             | deletion_date      
+-----------------+---------------------------------------------------------------------------------------------------------------------------------+----------------------------------+--------------------
| turbottest79983 | /subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983 | Microsoft.KeyVault/deletedVaults | 2021-08-13 10:44:31
| turbottest50436 | /subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest50436 | Microsoft.KeyVault/deletedVaults | 2021-08-13 10:44:18
+-----------------+---------------------------------------------------------------------------------------------------------------------------------+----------------------------------+--------------------
> select
  name,
  id,
  type,
  deletion_date,
  scheduled_purge_date
from
  azure_key_vault_deleted_vault
where
  scheduled_purge_date > (current_date - interval '1' day);
+-----------------+---------------------------------------------------------------------------------------------------------------------------------+----------------------------------+--------------------
| name            | id                                                                                                                              | type                             | deletion_date      
+-----------------+---------------------------------------------------------------------------------------------------------------------------------+----------------------------------+--------------------
| turbottest50436 | /subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest50436 | Microsoft.KeyVault/deletedVaults | 2021-08-13 10:44:18
| turbottest79983 | /subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/turbottest79983 | Microsoft.KeyVault/deletedVaults | 2021-08-13 10:44:31
+-----------------+---------------------------------------------------------------------------------------------------------------------------------+----------------------------------+--------------------

```
</details>

### CODE REVIEW

File: azure/plugin.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"azure_data_factory_pipeline":                                 tableAzureDataFactoryPipeline(ctx),
 			"azure_data_lake_analytics_account":                           tableAzureDataLakeAnalyticsAccount(ctx),
 			"azure_data_lake_store":                                       tableAzureDataLakeStore(ctx),
+			"azure_key_vault_deleted_vault":                               tableAzureKeyVaultDeletedVault(ctx),
```
Comment (khushboo9024) [inline azure/plugin.go:61]: please sort the list.

File: docs/tables/azure_key_vault_deleted_vault.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+select
+  name,
+  id,
+  type,
+  deletion_date,
+  scheduled_purge_date
+from
+  azure_key_vault_deleted_vault
+where
+scheduled_purge_date > (current_date - interval '1' day);
```
Comment (khushboo9024) [inline docs/tables/azure_key_vault_deleted_vault.md:32]:
```suggestion
  scheduled_purge_date > (current_date - interval '1' day);
```

File: docs/tables/azure_key_vault_deleted_vault.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  name,
+  id,
+  type,
+  deletion_date,
+  scheduled_purge_date
+from
+  azure_key_vault_deleted_vault;
+```
+
+### List vaults with scheduled purge date more than 1 day
```
Comment (khushboo9024) [inline docs/tables/azure_key_vault_deleted_vault.md:20]:
```suggestion
### List deleted vaults with scheduled purge date more than 1 day
```

**Review by khushboo9024 — COMMENTED** [review]
Please run the integration test again and add logs as the table name got modified.

**Review by khushboo9024 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**